contracts/EscrowedNori.sol

```diff
@@ -0,0 +1,725 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.13;
+
+import "@openzeppelin/contracts-upgradeable/utils/math/MathUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC777/IERC777RecipientUpgradeable.sol";
+import "./ERC777PresetPausablePermissioned.sol";
+import "./BridgedPolygonNORI.sol";
+import "./Removal.sol";
+import {RemovalUtils} from "./RemovalUtils.sol";
+
+/*
+Open Questions:
+- do we need to set up an escrow schedule with the full amount associated with the agreement?
+  or is it ok that the amount of the escrow schedule actually changes over time either with the addition
+  of new NORI or with the revocation of some?
+  - seems to me that the amount changing gradually makes more sense, because if a supplier only sells one of their
+  removals we actually want that single amount to release gradually over 10 years so that we have recourse to reclaim
+  those insurance tokens on behalf of that removal at some point in the future
+
+- do we need to allow specification of end time? or perhaps of duration? or can we assume always 10 years
+  and use that hardcoded constant?
+
+  - do we need any ability to create escrow schedules before tokens are received on behalf of a removal?
+      I would argue that we don't, and that the only placed escrow schedules should be created is in tokensReceived > depositFor
+
+  - should we set this up so that if a token is deposited on behalf of a removal id that DOESN'T have an entry for its schedule
+  -   startTime that we use the year (in epoch seconds) of the removal's vintage as the startime? it would allow us to migrate
+  -   seamlessly to the far superior world where the escrow schedules aren't so arbitrary and actually represent exactly
+  -   10 years for each removal.
+
+  - When it comes to keeping track of tokens that have been revoked, is it sufficient to keep a running total for any given escrow schedule,
+  and then to rely on events for more granular examination at the level of removalid and to see the times at which things were revoked?
```

Comment:
definitely-- I think we only track revocations in LockedNORI so that it makes it easier to run diffs against  on-chain vs the grants csv